### ISSUE TITLE
 DatePicker in antd version 4.2.4 is not compatible with version 2.26.0 of Moment.

### ISSUE THREAD

**Issue by milban** [ant-design/ant-design#24346]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/relaxed-cori-obmt7?file=/src/App.tsx)

### Steps to reproduce
1. install antd 4.2.4
2. install moment 2.26.0
3. import DatePicker from antd
4. declare const [date, setDate] = useState<Moment>();
5. set (value: Moment, _: string) => setDate(value) anonymous fn to DatePicker onChange props
6. type error appears

### What is expected?
Normally, the function is assigned to props.
No type error

### What is actually happening?
type error
```
Type '(value: Moment, _: string) => void' is not assignable to type '(value: Moment | null, dateString: string) => void'.
  Types of parameters 'value' and 'value' are incompatible.
    Type 'Moment | null' is not assignable to type 'Moment'.
      Type 'null' is not assignable to type 'Moment'.ts(2322)
Picker.d.ts(43, 5): The expected type comes from property 'onChange' which is declared here on type '(IntrinsicAttributes & IntrinsicClassAttributes<Component<PickerProps<Moment>, any, any>> & Readonly<...> & Readonly<...>) | (IntrinsicAttributes & ... 2 more ... & Readonly<...>) | (IntrinsicAttributes & ... 2 more ... & Readonly<...>)'
```

### Curren Solution
Downgrade moment version to 2.25.3

| Environment | Info |
|---|---|
| antd | 4.2.4 |
| React | 16.13.1 |
| System | MacOS |
| Browser | Chrome |

---
It takes some time for the error to appear in the DatePicker in codesandbox.

**Comment by zombieJ** [comment]
Duplicate of #24322

**Comment by abenhamdine** [comment]
I don't think it's fixed, I still have this issue with antd 4.30 and momentjs 2.26

**Comment by zombieJ** [comment]
hi @abenhamdine, try to clean all lock file and reinstall node modules. If you still meet this, pls feel free to create a new issue with reproduce.
